conbench/entities/benchmark_result.py

```diff
@@ -507,7 +459,115 @@ def ui_hardware_short(self) -> str:
+    result_data_for_db = stats_usergiven.copy()
+
+    if len(samples) > 1:
+        # There is a special case where for one sample the iteration count
+        # may be larger than one, see below.
+        if stats_usergiven["iterations"] != len(samples):
+            raise BenchmarkResultValidationError(
+                f'iterations count ({ stats_usergiven["iterations"] }) does '
+                f"not match sample count ({len(samples)})"
+            )
+
+    if len(samples) >= 3:
+        # See https://github.com/conbench/conbench/issues/802 and
+        # https://github.com/conbench/conbench/issues/1118
+        q1, q3 = np.percentile(samples, [25, 75])
+
+        aggregates = {
+            "q1": q1,
+            "q3": q3,
+            "mean": np.mean(samples),
+            "median": np.median(samples),
+            "min": np.min(samples),
+            "max": np.max(samples),
+            # https://numpy.org/doc/stable/reference/generated/numpy.std.html
+            # This has N in the divisor (ddof=0)
+            "stdev": np.std(samples),
+            "iqr": q3 - q1,
+        }
+
+        # Now, overwrite with self-derived aggregates:
+        for key, value in aggregates.items():
+            result_data_for_db[key] = sigfig.round(value, sigfigs=5)
+
+            # Log upon conflict. Lett the automatically derived value win,
+            # to achieve consistency between the provided samples and the
+            # aggregates.
+            if key in stats_usergiven:
+                if not floatcomp(stats_usergiven[key], value):
+                    log.warning(
+                        "key %s, user-given val %s vs. calculated %s",
+                        key,
+                        stats_usergiven[key],
+                        value,
+                    )
+
+    if len(samples) == 1:
+        if stats_usergiven["iterations"] == 1:
+            # If user provides aggregates, then it's unclear what they
+            # mean.
+            for k in ("q1", "q2", "mean", "median", "min", "max", "stdev", "iqr"):
+                if stats_usergiven.get(k) is not None:
+                    raise BenchmarkResultValidationError(
+                        f"one data point from one iteration: property `{k}` "
+                        "is unexpected"
+                    )
+
+        if stats_usergiven["iterations"] > 1:
+            # Note(JP): Do not yet handle this in a special way, but I think
+            # that this here is precisely the one way to report the result of a
+            # microbenchmark with cross-repetition stats: data: length 1 (the
+            # duration of running many repetitions), and iterations: a number
+            # larger than 1, while mean, max, are now set (then these can be
+            # assumed to be derived from _within_ the microbenchmark. Note that
+            # this is a very special condition). The mean time for a single
+            # repetition now is sample/iterations.
+            ...
```

Comment:
While I was here, I thought that this is a special condition. Not changing behavior yet.
@alistaire47's thoughts:  https://github.com/conbench/conbench/pull/1119#discussion_r1167300853